handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandlerBuilder.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.proxy;
+
+import static io.netty.util.internal.ObjectUtil.checkNotNull;
+
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+
+import io.netty.util.internal.StringUtil;
+
+/**
+ * Creates a new {@link ProxyHandler} of desired type with requested proxy
+ * address and authentication. The builder gracefully accepts null or empty
+ * values for username and password, which allows callers to pass in values from
+ * configuration files without additional checking.
+ */
+public final class ProxyHandlerBuilder {
+
+  public static enum ProxyType {
+    HTTP, SOCKS4, SOCKS5
```

Comment:
i am confused, could you give me a clear signal for the one or the other? thanks :)